src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -228,11 +226,11 @@ void BaseCouplingScheme::advance()
         // coupling iteration.
         PRECICE_ASSERT(math::greater(_computedTimeWindowPart, 0.0));
         _time = _time - _computedTimeWindowPart;
-        _timeWindows -= 1;
       } else { // write output, prepare for next window
```

Comment:
Now, you treat the time window counter and time differently. I guess this could reduce readability again.

Reply:
True. So we should probably also touch `_time` and make it consistent.